tests/subsys/fs/nvs/src/main.c

```diff
@@ -244,6 +244,109 @@ void test_nvs_gc(void)
 	}
 }
 
+static int flash_sim_erase_calls_find(struct stats_hdr *hdr, void *arg,
+				      const char *name, uint16_t off)
+{
+	if (!strcmp(name, "flash_erase_calls")) {
+		u32_t **flash_erase_stat = (u32_t **) arg;
+		*flash_erase_stat = (u32_t *)((u8_t *)hdr + off);
+	}
+
+	return 0;
+}
+
+static int flash_sim_max_erase_calls_find(struct stats_hdr *hdr, void *arg,
+					  const char *name, uint16_t off)
+{
+	if (!strcmp(name, "max_erase_calls")) {
+		u32_t **max_erase_calls = (u32_t **) arg;
+		*max_erase_calls = (u32_t *)((u8_t *)hdr + off);
+	}
+
+	return 0;
+}
+
+static int flash_sim_max_len_find(struct stats_hdr *hdr, void *arg,
+				  const char *name, uint16_t off)
+{
+	if (!strcmp(name, "max_len")) {
+		u32_t **max_len = (u32_t **) arg;
+		*max_len = (u32_t *)((u8_t *)hdr + off);
+	}
+
+	return 0;
+}
+
+void test_nvs_corrupted_sector_close_operation(void)
+{
+	int err;
+	int len;
+	u8_t buf[32];
+	u8_t rd_buf[32];
+	u32_t *flash_write_stat;
+	u32_t *flash_erase_stat;
+	u32_t *flash_max_write_calls;
+	u32_t *flash_max_erase_calls;
+	u32_t *flash_max_len;
+
+	const u16_t max_id = 10;
+	/* 25th write will trigger GC. */
+	const u16_t max_writes = 26;
+
+	/* Get the address of simulator parameters. */
+	stats_walk(sim_thresholds, flash_sim_max_write_calls_find,
+		   &flash_max_write_calls);
+	stats_walk(sim_thresholds, flash_sim_max_erase_calls_find,
+		   &flash_max_erase_calls);
+	stats_walk(sim_thresholds, flash_sim_max_len_find,
+		   &flash_max_len);
+	stats_walk(sim_stats, flash_sim_write_calls_find, &flash_write_stat);
+	stats_walk(sim_stats, flash_sim_erase_calls_find, &flash_erase_stat);
+
+	err = nvs_init(&fs, DT_FLASH_DEV_NAME);
+	zassert_true(err == 0,  "nvs_init call failure: %d", err);
+
+	for (u16_t i = 0; i < max_writes; i++) {
+		u8_t id = (i % max_id);
+		u8_t id_data = id + max_id * (i / max_id);
+
+		memset(buf, id_data, sizeof(buf));
+
+		if (i == max_writes - 1) {
+			/* Reset stats. */
+			*flash_write_stat = 0;
+			*flash_erase_stat = 0;
+
+			/* Block write calls and simulate power down
+			 * during sector closing operation.
+			 */
+			*flash_max_write_calls = 1;
+			*flash_max_erase_calls = 1;
+			*flash_max_len = 4;
+		}
+
+		len = nvs_write(&fs, id, buf, sizeof(buf));
+		zassert_true(len == sizeof(buf), "nvs_write failed: %d", len);
+	}
+
+	err = nvs_init(&fs, DT_FLASH_DEV_NAME);
+	zassert_true(err == 0,  "nvs_init call failure: %d", err);
+
+	for (u16_t id = 0; id < max_id; id++) {
+		len = nvs_read(&fs, id, rd_buf, sizeof(buf));
+		zassert_true(len == sizeof(rd_buf),
+			     "nvs_read unexpected failure: %d", len);
+
+		for (u16_t i = 0; i < sizeof(rd_buf); i++) {
+			rd_buf[i] = rd_buf[i] % max_id;
+			buf[i] = id;
+		}
+		zassert_mem_equal(buf, rd_buf, sizeof(rd_buf),
+				"RD buff should be equal to the WR buff");
+
+	}
```

Comment:
can you add one more write and read operation after that to ensure NVS is capable to do so?